src/jit/codegenxarch.cpp

```diff
@@ -2176,14 +2176,25 @@ void CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
                     break;
                 }
 #endif // !defined(_TARGET_64BIT_)
+                
+#ifdef FEATURE_SIMD
+                if (varTypeIsSIMD(targetType) && (targetReg != REG_NA) && op1->IsCnsIntOrI())
```

Comment:
No, this won't actually currently happen - it is sort of an "anticipatory" change for 1st class structs. I haven't run all the diffs on this exact set of changes, but an earlier (very close) iteration of this got zero diffs on all desktop targets, plus all of the coreclr assemblies & tests that successfully ran with jit_diffs. I'm still in the "zero diffs" phase of this work.

---
In reply to: [75382478](https://github.com/dotnet/coreclr/pull/6780#discussion_r75382478) [](ancestors = 75382478)